### ISSUE TITLE
URL verifier not accepting photo

### ISSUE THREAD

**Issue by youkoso** [electerious/Lychee#482]
Hello! I'm trying to import some photos from URL that ending with ".jpg:orig" or ".png?cache_key=asdasd" but Lychee don't accept this.
Can you make smarter URL validation (or option to disable this)?

Thank you! Awesome project.

**Comment by qligier** [comment]
Hello,

The [getExtension](/electerious/Lychee/blob/master/php/modules/misc.php#L118) function should use the pathinfo function: 
```php
$extension = pathinfo($filename, PATHINFO_EXTENSION);
```

URLs should also be processed through parse_url to get rid of all unnecessary components:
```php
var_dump(pathinfo(parse_url('http://www.google.com/image.png?cache_key=asdasd', PHP_URL_PATH), PATHINFO_EXTENSION));
# string(3) "png"
```

Note that this doesn't solve the _.jpg:orig_ case.
Quentin

**Comment by electerious** [comment]
@qligier Thanks for the suggestion. Using `pathinfo` and `parse_url` are welcome changes. Feel free to open a PR for this :)

**Comment by qligier** [comment]
Will do. Do you prefer the PRs to be on the dev branch? Is it ok if I open one PR for multiple changes or do you want one PR per feature? 

Quentin

**Comment by electerious** [comment]
One PR per feature and all based on the develop branch. Thanks!

**Comment by qligier** [comment]
In what cases does the URL end with ":orig"? Is it worth adding a test for this in the code?

**Comment by youkoso** [comment]
@qligier ":orig" is original size of images in Twitter. For example: https://pbs.twimg.com/media/CdBWRt5XEAAS8mG.jpg and 
https://pbs.twimg.com/media/CdBWRt5XEAAS8mG.jpg:orig
Not necessary, but would be perfect. Or maybe another way to import images from Twitter?

**Comment by qligier** [comment]
Ok, we could add some lines of code that deal with this (and possibly others) special URL.

**Comment by youkoso** [comment]
Thank you!

**Comment by qligier** [comment]
Solved with #490.

**Comment by electerious** [comment]
Importing photo with an extension containing a ':' will be possible in version 3.1. Thanks @qligier for the implementation.

**Comment by electerious** [comment]
Added support for photo URLs containing a question mark: https://github.com/electerious/Lychee/commit/fbab70f244665a1a216731ce4aea4d9f8b233cd8

**Comment by electerious** [comment]
@qligier `parse_url($filename, PHP_URL_PATH)` should normally not contain the query and fragment, but I got an error without the [new explode](https://github.com/electerious/Lychee/commit/fbab70f244665a1a216731ce4aea4d9f8b233cd8). Need to do some additional tests.

**Comment by qligier** [comment]
Solved with #493. The script was trying to create a file with the question mark in it (it's forbidden in Windows and forbidden unescaped in Unix).

**Comment by electerious** [comment]
@qligier Works! I removed my newly added `explode`.
